pandas/core/generic.py

```diff
@@ -1940,6 +1940,84 @@ def tail(self, n=5):
             return self
         return self.iloc[-n:]
 
+  
+    def sample(self, n = None, frac = None , replace = False, weights = None, seed = None):
+        """
+        Returns a sample of rows from object. 
+    
+        Note that .sample() (with no arguments) will return five random rows.
+    
+        Parameters
+        ----------
+            n: Number of rows to return. Cannot be used with frac.
+               Default = 5 if frac = None. 
+            frac: share of rows to return. Cannot be used with n. 
+            replace {True, False}: Sample with or without replacement.
+            weights: Accepts a Series, list, or ndarray of weights. If called on a 
+                    DataFrame, will also accept the name of a column as a string. 
+                    Must be same length as index.  
+                    Default 'None' results in equal probability weighting.
+            seed: seed to be fed to numpy random.RandomState() Function. Default None. 
+        -----------
+        Note: sample sizes are passed to numpy.random.choice(), which accepts negative values. 
+              
+              If a negative value is passed, this will be interpreted with respect to 
+              number of rows. So n = -1 will result in a sample with 
+              [original number of rows - 1] rows.
+              
+              A similar behavior applies to values passed to frac. A value of frac = -0.75 
+              will return a sample with a size of [0.25 * original number of rows] rows, 
+              give or take rounding. 
+              
+              Finally, note that non-integer arguments passed to n will be rounded off by numpy. 
+              So n = 3.1 will return 3 rows, and n = 3.9 rows will return 4 rows. 
+        """
+    
+        # Setup vars
+        rs = np.random.RandomState(seed)
+        length = len(self)
```

Comment:
I don't think much is gained by making this a separate variable -- I would probably still use `len(self)` throughout (just my two-cents).